### ISSUE TITLE
About the result of PPO

### ISSUE THREAD

**Issue by chen-xihui** [ShengrenHou/DRL-for-Energy-Systems-Optimal-Scheduling#8]
We set the parameters as you mentioned in the paper. The mean reward of 5 episodes is shown as below.
![image](https://github.com/ShengrenHou/DRL-for-Energy-Systems-Optimal-Scheduling/assets/64118984/5ef130e1-b89a-4247-891c-b4e06a44fb01)
As can be seen in the figure, if we train 1000 episodes the mean reward will  approach -60, which is really bad.
We also train 20000 episodes for each seed. The result can be seen below.
![image](https://github.com/ShengrenHou/DRL-for-Energy-Systems-Optimal-Scheduling/assets/64118984/9949635d-d60b-40dd-98e4-856567efba6d)
The mean reward of 500 episodes finally approaches -20. While the results of training process in your paper is so nice, I wonder what makes the difference? And what are other people's results of the reproduction of the paper？
To be honest, now i want a relatively good results as a baseline. Do you think the mean reward shown above can been seen as a baseline? I would be grateful if you could solve my confusion.

**Comment by chen-xihui** [comment]
We set episode=2000, learning_rate=5e-6 and the results are as followed.
1. The mean reward of 5 episodes.
![image](https://github.com/ShengrenHou/DRL-for-Energy-Systems-Optimal-Scheduling/assets/64118984/0aa5d208-6703-494e-852d-a1051eacb42f)
2. The results of optimization information are as followed.
![image](https://github.com/ShengrenHou/DRL-for-Energy-Systems-Optimal-Scheduling/assets/64118984/85da02dd-64b3-4957-820a-90efb49f9317)
3. The results of Evoluation information are as followed.
![image](https://github.com/ShengrenHou/DRL-for-Energy-Systems-Optimal-Scheduling/assets/64118984/d654b8ef-01a4-46fc-b13d-05b5a09c59fb)

**Comment by chen-xihui** [comment]
We also set episode=5000, learning_rate=5e-6 and the results are as followed.
1. The mean reward of 30 episodes.
![image](https://github.com/ShengrenHou/DRL-for-Energy-Systems-Optimal-Scheduling/assets/64118984/467d6208-1dbd-4e7d-a35a-45cd9d440998)
2. The results of optimization information are as followed.
![image](https://github.com/ShengrenHou/DRL-for-Energy-Systems-Optimal-Scheduling/assets/64118984/d4ffe021-95b4-4b16-9132-9641f58afd1c)
3. The results of Evoluation information are as followed.
![image](https://github.com/ShengrenHou/DRL-for-Energy-Systems-Optimal-Scheduling/assets/64118984/476f0107-c5ca-4b54-86b6-04e9d28d00f5)

DO you think the result can be seen as a baseline?

**Comment by chen-xihui** [comment]
It seems it is impossible to get a good result within 1000 episodes or 2000 episodes.

**Comment by chen-xihui** [comment]
Look forward to hearing from you.

**Comment by ShengrenHou** [comment]
hey, there. Sorry for the late reply. I was on holiday last week. 
The current result you get from PPO can not be counted as a baseline. 
The result is not good enough. According to my experience, at least the algorithm should learn to discharge to SOC (0.2) at the end of the operation to earn money. 
I also don't know why you can not reproduce the result. Another PhD candidate in China also report the problem for the current PPO, but I still don't have time to check it. Maybe I uploaded the wrong version of PPO. But he validated that DDPG and TD3 can work well in the environment I modeled.   
Try more on DDPG or TD3. 
Best regards

**Comment by chen-xihui** [comment]
Look forward to hearing from you.

**Comment by ShengrenHou** [comment]
Hey,
The environment works well. I will check the implementation of PPO after Sept.
Before that, you can 1. use DDPG or TD3, 2. Implement a new PPO by yourself. There are a lot of resources implementing clean PPO. 
Best regards,

**Comment by chen-xihui** [comment]
sir, sorry for disturbing you. Have you checked your code of PPO?&nbsp;







&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "ShengrenHou/DRL-for-Energy-Systems-Optimal-Scheduling"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年6月9日(星期五) 下午3:24
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [ShengrenHou/DRL-for-Energy-Systems-Optimal-Scheduling] About the result of PPO (Issue #8)





 
Hey,
 The environment works well. I will check the implementation of PPO after Sept.
 Before that, you can 1. use DDPG or TD3, 2. Implement a new PPO by yourself. There are a lot of resources implementing clean PPO.
 Best regards,
